runtime/src/accounts.rs

```diff
@@ -286,6 +290,28 @@ impl Accounts {
         }
     }
 
+    fn account_shared_data_from_program(
+        key: &Pubkey,
+        program: &LoadedProgram,
+        program_accounts: &HashMap<Pubkey, &Pubkey>,
+    ) -> Result<AccountSharedData> {
+        // Check for tombstone
+        if program.is_tombstone() {
+            return Err(TransactionError::InvalidProgramForExecution);
```

Comment:
I think this changes the error code for attempting to execute a closed program.

Reply:
I'll check and fix it.

Reply:
We probably need to store the reason for tombstone in the cache. Otherwise we don't have enough information to return the correct error case.